AsyncDisplayKit/ASCollectionData.h

```diff
@@ -0,0 +1,131 @@
+                           block:(__attribute((noescape)) void(^)(ASCollectionData * data))block;
+
+/**
+ * Adds an item to the current section. If this method is called outside of an addSectionWithIdentifier:
+ * block, a default section will be created and used.
+ *
+ * @param identifier The identifier for the new item.
+ * @param nodeBlock A block that will be used to construct the node for the item.
+ *
+ * @note If an item already exists with this identifier, the node block will be ignored.
+ * @note It is more performant to specify the node block inline here, rather than calling a method that generates it, but the difference is not prohibitive.
+ */
+- (void)addItemWithIdentifier:(ASItemIdentifier)identifier
+                    nodeBlock:(ASCellNodeBlock)nodeBlock;
+
+/**
+ * Adds a supplementary element to the current section. If this method is called outside of an addSectionWithIdentifier:
+ * block, a default section will be created and used.
+ *
+ * @param elementKind The kind of supplementary element to add.
+ * @param identifier The identifier for the new item.
+ * @param index The index for the supplementary element. This can be @c NSNotFound.
+ * @param nodeBlock A block that will be used to construct the node for the supplementary element.
+ *
+ * @note If an element already exists with this identifier, the node block will be ignored.
+ * @note It is more performant to specify the node block inline here, rather than calling a method that generates it, but the difference is not prohibitive.
+ */
+- (void)addSupplementaryElementOfKind:(NSString *)elementKind
+                       withIdentifier:(ASItemIdentifier)identifier
+                                index:(NSInteger)index
+                            nodeBlock:(ASCellNodeBlock)nodeBlock;
+
+/**
+ * Finds or creates an item with the given identifier.
+ *
+ * @param identifier The identifier for the item.
+ * @param nodeBlock A block that will be used to construct the node for the item.
+ *
+ * You can use this method to get more fine-grained control over the collection data.
+ * The returned item can be inserted into the @c mutableItems of an @c ASCollectionSection.
+ *
+ * Note that if an item already exists with this identifier, the node block will be ignored.
+ * @note It is more performant to specify the node block inline here, rather than calling a method that generates it, but the difference is not prohibitive.
+ */
+- (id<ASCollectionItem>)itemWithIdentifier:(ASItemIdentifier)identifier
+                                 nodeBlock:(ASCellNodeBlock)nodeBlock;
+
+/**
+ * Finds or creates a section with the given identifier.
+ *
+ * @param identifier The identifier for the section.
+ *
+ * You can use this method to get more fine-grained control over the collection data.
+ * The returned section can be inserted into @c mutableSections.
+ */
+- (id<ASCollectionSection>)sectionWithIdentifier:(ASSectionIdentifier)identifier;
```

Comment:
Is this more like sectionForIdentifier: ?  This sounds a bit like a generator method that would create a section given an identifier, even if it didn't exist previously.  Not definitive feedback though.